### PR TITLE
Add volumes, resource limits, endpoints, and envs

### PR DESCRIPTION
**Pull request by ericwill**
Signed-off-by: Eric Williams <ericwill@redhat.com>

### What does this PR do?
Adds the volume, endpoint, env, and resource limit entries to the `che-theia-plugins.yaml` file.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
Part of eclipse/che#18215

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: che-theia-plugins.yaml [hunk 1]
```diff
@@ -130,6 +162,10 @@ plugins:
       revision: v1.5.0
     sidecar:
       directory: java8
+      memoryLimit: "1500Mi"
+      volumeMounts:
+      - name: m2
+        mount: "/home/theia/.m2"
```
Comment (benoitf) [inline che-theia-plugins.yaml:168]: I think it should be `path`

File: che-theia-plugins.yaml [hunk 2]
```diff
@@ -219,20 +281,44 @@ plugins:
       - redhat/mta
     sidecar:
       image: quay.io/windupeng/mta-vscode-extension:latest
+      name: mta-extension
+      memoryLimit: 1500Mi
+      mountSources: true
+      ports:
```
Comment (benoitf) [inline che-theia-plugins.yaml:287]: `endpoints`

Comment (benoitf) [inline che-theia-plugins.yaml:287]: I think we can remove `ports` as it duplicates `endpoints`

File: che-theia-plugins.yaml (hunk 1 above)
Comment (ericwill) [reply che-theia-plugins.yaml:168]: Yes I messed up my find/replace :smiley:

**Review by benoitf — APPROVED** [review]
👍

**Comment by ericwill** [comment]
Looks like GitHub action is having an outage of some kind: https://www.githubstatus.com/incidents/sxthxvdlljk4

**Comment by benoitf** [comment]
I've restarted jobs

**Pull request merged**